src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -39,17 +39,17 @@ internal class TagHelperParseTreeRewriter
         private readonly string _tagHelperPrefix;
         private readonly List<KeyValuePair<string, string>> _htmlAttributeTracker;
         private readonly StringBuilder _attributeValueBuilder;
-        private readonly TagHelperDescriptorProvider _provider;
+        private readonly TagHelperBinder _tagHelperBinder;
         private readonly Stack<TagBlockTracker> _trackerStack;
         private readonly Stack<BlockBuilder> _blockStack;
         private TagHelperBlockTracker _currentTagHelperTracker;
         private BlockBuilder _currentBlock;
         private string _currentParentTagName;
 
-        public TagHelperParseTreeRewriter(string tagHelperPrefix, TagHelperDescriptorProvider provider)
+        public TagHelperParseTreeRewriter(string tagHelperPrefix, TagHelperBinder tagHelperBinder)
```

Comment:
think about passing in the descriptors here and then creating the `THB` in the constructor.